docs/src/main/asciidoc/kotlin.adoc

```diff
@@ -173,6 +173,203 @@ for example, then `<option>all-open:annotation=javax.enterprise.context.Applicat
 
 Future versions of Quarkus will configure the Kotlin compiler plugin in a way that will make it unnecessary to alter this configuration.
 
+== Important Gradle configuration points
+
+Similar to the Maven configuration, when using Gradle, the following modifications are required when Kotlin is selected:
+
+* The `quarkus-kotlin` artifact is added to the dependencies. This artifact provides support for Kotlin in the live reload mode (more about this later on)
+* The `kotlin-stdlib-jdk8` is also added as a dependency.
+* The Kotlin plugin is activated, which implicitly adds `sourceDirectory` and `testSourceDirectory` build properties to point to Kotlin sources (`src/main/kotlin` and `src/test/kotlin` respectively)
+* All open Kotlin plugin tells the compiler not to mark as final, those classes with the annotations highlighted (customize as required)
+* When using native-image, the use of http (or https) protocol(s) must be declared
+* An example configuration follows:
+
+[source,groovy,subs=attributes+]
+----
+group = '...' // set your group
+version = '1.0.0-SNAPSHOT'
+
+plugins {
+    id 'java'
+    id 'io.quarkus' version '{quarkus-version}' // <1>
+
+    id "org.jetbrains.kotlin.jvm" version "{kotlin-version}" // <2>
+    id "org.jetbrains.kotlin.plugin.allopen" version "{kotlin-version}" // <2>
+}
+
+apply plugin: 'io.quarkus'
+
+repositories {
+    mavenCentral()
+}
+
+dependencies { 
+    implementation 'org.jetbrains.kotlin:kotlin-stdlib-jdk8:{kotlin-version}'
+
+    // <3>
+    implementation enforcedPlatform('io.quarkus:quarkus-bom:{quarkus-version}')
+    implementation 'io.quarkus:quarkus-resteasy'
+    implementation 'io.quarkus:quarkus-resteasy-jsonb'
+    implementation 'io.quarkus:quarkus-kotlin'
+
+    testImplementation 'io.quarkus:quarkus-junit5'
+    testImplementation 'io.rest-assured:rest-assured:{rest-assured-version}'  // <4>
+}
+
+sourceCompatibility = '1.8'
+targetCompatibility = '1.8'
+
+test {
+    useJUnitPlatform()
```

Comment:
Interesting! Can you explain a little why this is needed?

Reply:
I assume you are referring to the JUnit activation?  It's not activated by default, despite the dependency being added.  I have the native test exclude as they fail with Gradle (yet work with Maven).

Reply:
Yeah, JUnit is what I had in mind. Is there a Gradle link that mentions the need to add this?

Reply:
https://docs.gradle.org/current/userguide/java_testing.html#using_junit5

It is also referenced in the Quarkus Gradle plugin page.